internal/proxy: TestClientLimitsMaxConnections failed

This test failed!

To configure my behavior, see [the Flaky Bot documentation](https://github.com/googleapis/repo-automation-bots/tree/main/packages/flakybot).

If I'm commenting on this issue too often, add the `flakybot: quiet` label and
I will stop commenting.

---

commit: 19d7ad969bce68d2b709e5f09f2bcf6869e224c5
buildURL: https://github.com/GoogleCloudPlatform/alloydb-auth-proxy/actions/runs/7550125554
status: failed

=== RUN   TestClientLimitsMaxConnections
2024/01/17 02:14:28 [proj.region.clust.inst] Listening on 127.0.0.1:5000
2024/01/17 02:14:28 [proj.region.clust.inst] accepted connection from 127.0.0.1:49207
2024/01/17 02:14:28 [proj.region.clust.inst] accepted connection from 127.0.0.1:49206
2024/01/17 02:14:28 max connections (1) exceeded, refusing new connection
    proxy_test.go:347: dial attempts did not match expected, want = 1, got = 0
--- FAIL: TestClientLimitsMaxConnections (0.00s)

--- a/internal/proxy/proxy_test.go
+++ b/internal/proxy/proxy_test.go
@@ -342,10 +342,19 @@ func TestClientLimitsMaxConnections(t *testing.T) {
 	// it doesn't matter which is closed
 	wantEOF(t, conn1, conn2)
 
-	want := 1
-	if got := d.dialAttempts(); got != want {
+	tryDialAttempts := func(t *testing.T, want int) {
+		var got int
+		for i := 0; i < 10; i++ {
+			got = d.dialAttempts()
+			if got == want {
+				return
+			}
+			time.Sleep(100 * time.Millisecond)
+		}
 		t.Fatalf("dial attempts did not match expected, want = %v, got = %v", want, got)
 	}
+	want := 1
+	tryDialAttempts(t, want)
 }
 
 func tryTCPDial(t *testing.T, addr string) net.Conn {

